### DISCUSSION TITLE
vscode: How do I direct julia code execution to a Julia Jupyter Kernel Interactive Window

### DISCUSSION THREAD

**Discussion by pjcrosbie** [quarto-dev/quarto-cli#9180] [Q&A]
### Description

````
---
title: "Simple Demo"
format:
  pdf:
    code-fold: true
jupyter: julia-1.10.2
---
````

#### vscode: How do I direct julia code execution to a Julia Jupyter Kernel Interactive Window?

Apologies in advance if this primarily a vscode question.

In vscode, I can open an "Interactive" window using "Jupyter: Open a new interactive window". I can then execute julia code in this window directly using the interactive window's "execute code" input box. 

However, when I select "Run Cell..." in the qmd file, the code is execute in a julia repl started in vscode's terminal window.

How do I direct the "Run Cell..." code execution to the Jupyter Interactive window instead of the terminal repl?

I note that executing in a Jupyter Interactive window appears to be the default behaviour for a quarto document using a python kernel.

````{julia}
f(x) = x^2
````

I appreciate that I can use `quarto preview mydoc.qmd` to get a live preview but I don't find the preview experience is not as nice as the vscode builtin Jupyter Interactive window.

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
